keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resolvers/ImageResolver.kt

```diff
@@ -44,46 +47,46 @@ class ImageResolver(
     val imageProvider = resource.spec.imageProvider ?: return resource
     val image = runBlocking {
       when (imageProvider) {
-        is ArtifactImageProvider -> resolveFromArtifact(resource, imageProvider)
+        is ReferenceArtifactImageProvider -> resolveFromReference(resource, imageProvider)
+        // todo eb: artifact provider is here for backwards compatibility. Remove?
+        is ArtifactImageProvider -> resolveFromArtifact(resource, imageProvider.deliveryArtifact as DebianArtifact)
```

Comment:
Would be good to get rid of this if we can

Reply:
Ya I agree. We have to do it after migrating all current delivery configs though.